src/main/java/com/programmers/springbootbasic/domain/voucher/domain/entity/Voucher.java

```diff
@@ -0,0 +1,39 @@
+package com.programmers.springbootbasic.domain.voucher.domain.entity;
+
+
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherType.VoucherType;
+import java.util.UUID;
+
+public class Voucher {
+    private UUID id;
+    private VoucherType voucherType;
+    private Integer benefitValue;
+
+    public Voucher(UUID id, VoucherType voucherType, Integer benefitValue) {
+        this.id = id;
+        this.voucherType = voucherType;
+        this.benefitValue = benefitValue;
+    }
+
+    public Voucher(VoucherType voucherType, Integer benefitValue) {
+        this.id = UUID.randomUUID();
+        this.voucherType = voucherType;
+        this.benefitValue = benefitValue;
+    }
+
+    public Voucher() {
+    }
```

Comment:
```suggestion
    public Voucher() {
    }

    public Voucher(VoucherType voucherType, Integer benefitValue) {
        this(UUID.randomUUID(), voucherType, benefitValue);
    }

    public Voucher(UUID id, VoucherType voucherType, Integer benefitValue) {
        this.id = id;
        this.voucherType = voucherType;
        this.benefitValue = benefitValue;
    }
```

생성자 오버로딩을 사용하는 경우 인자 값이 가장 많이 필요한 메서드가 대부분 '필드 초기화를 처리하는' 생성자이곤 합니다.
따라서 this()를 사용하여 오버로딩으로 초기화하는 부분을 한 곳에서 처리할 수 있도록 하여 영향도를 낮추는 것이 좋습니다.

Reply:
[b02f208](https://github.com/prgrms-be-devcourse/springboot-basic/pull/902/commits/b02f208cfde71ab38b4e085157cab5bfd09f8073)

영향도를 한번도 생각해 본적 이 없는데, 좋은 지적 감사합니다!